js/components/Confirmation.jsx

```diff
@@ -5,21 +5,219 @@ import ProgressButton from "us-forms-system/lib/js/components/ProgressButton";
+    }
+
+    // Instantiates an XHR request, then submits to submitUrl
+    make_xhr_request(formData, submitUrl, formContext = null) {
+        // console.log("API_URL: " + submitUrl);
+        // console.log("Appending formConfig to formData!");
+        // console.log(formData);
+
+        // Create the XHR request
+        var request = new XMLHttpRequest();
+        // Return it as a Promise
+        return new Promise(function (resolve, reject) {
+                // Setup our listener to process complete requests
+                request.onreadystatechange = function () {
+                    // Only run if the request is complete
+                    if (request.readyState !== 4) return;
+                        // Process the response
+                    if (request.status >= 200 && request.status < 300) {
+                        // If successful
+                        resolve(request);
+                    } else {
+                        // If failed
+                        reject({
+                                   status: request.status,
+                                   statusText: request.statusText
+                               });
+                    }
+                };
+
+            // Setup our HTTP request, headers & send our payload ...
+            request.open('POST', submitUrl, true);
+            request.setRequestHeader("Content-Type", "application/json;charset=UTF-8");
+
+            request.send(JSON.stringify(formData));
+            })
+            .then(function (response) {
+                formContext.markDone(formContext);
+                formContext.clearLocalStorage();
+                return response;
+            })
+            .catch(function (error) {
+                console.log("Error: ");
+                console.log(error);
+                // return error;
+            });
+    }
+
+    // Renders the right button based on the state formSubmissionState
+    // 0 = Submit (regular)
+    // 1 = Loading (three dots animation)
+    // 2 = Done (animated checkmark, green button)
+    renderButton() {
+        switch(this.state.formSubmissionState) {
+            case 1:
+                return <button className="confirmation-button" onClick={() => this.resubmitForm()}>
+                    <div className="confirmation-button--loading-btn"></div>
+                </button>;
+            case 2:
+                return <button className="confirmation-button confirmation-button__success" onClick={null}>
+                    <div className="confirmation-button--checkmark-container">
+                        <div className="confirmation-button--checkmark draw">&nbsp;</div>
+                    </div>
+                </button>;
+            default:
+                return <button className="confirmation-button" onClick={() => this.resubmitForm()}>Submit</button>;
+        }
+    }
+
+    render() {
+        let confirmationCaseNumber = this.state.confirmationCaseNumber;
```

Comment:
this is another nice to have, when we refactor I would prefer keeping the render component clean and simple, and moving these templates into their own components that just get referenced here